src/types.py

```diff
@@ -599,164 +823,96 @@ def unify_with_args(self, concrete: "FunctionSignature") -> bool:
         return can_unify
 
 
-def type_from_ctype(ctype: CType, typemap: TypeMap, array_decay: bool = True) -> Type:
-    real_ctype = resolve_typedefs(ctype, typemap)
-    if isinstance(real_ctype, ca.ArrayDecl):
-        inner_type, dim = array_type_and_dim(real_ctype, typemap)
-        if array_decay:
-            return Type.ptr(inner_type)
-        size = inner_type.get_size_bits()
-        if size is not None and dim is not None:
-            size *= dim
-        else:
-            size = None
-        return Type._ctype(real_ctype, typemap, size=size)
-    if isinstance(real_ctype, ca.PtrDecl):
-        return Type.ptr(type_from_ctype(real_ctype.type, typemap, array_decay=False))
-    if isinstance(real_ctype, ca.FuncDecl):
-        fn = parse_function(real_ctype)
-        assert fn is not None
-        fn_sig = FunctionSignature(
-            return_type=Type.void(),
-            is_variadic=fn.is_variadic,
-        )
-        if fn.ret_type is not None:
-            fn_sig.return_type = type_from_ctype(fn.ret_type, typemap)
-        if fn.params is not None:
-            fn_sig.params = [
-                FunctionParam(
-                    name=param.name or "",
-                    type=type_from_ctype(param.type, typemap),
-                )
-                for param in fn.params
-            ]
-            fn_sig.params_known = True
-        return Type.function(fn_sig)
-    if isinstance(real_ctype, ca.TypeDecl):
-        if isinstance(real_ctype.type, (ca.Struct, ca.Union)):
-            struct = parse_struct(real_ctype.type, typemap)
-            return Type._ctype(struct.type, typemap, size=struct.size * 8)
-        names = (
-            ["int"] if isinstance(real_ctype.type, ca.Enum) else real_ctype.type.names
+@dataclass(eq=False)
+class StructField:
+    type: Type
+    offset_bits: int
+    name: str
+
+
+@dataclass(eq=False)
+class StructDeclaration:
+    # TODO: Document
+    size_bits: int
+    align_bits: int
+    tag_name: Optional[str]
+    typedef_name: Optional[str]
+    fields: List[StructField]
+    has_bitfields: bool
+    is_union: bool
```

Comment:
I decided to measure everything here in bits instead of bytes, even though there isn't really support for bitfields yet. I figured this would make it easier to add support for bitfields in the future? Although it's confusing, I tried to compensate by being as explicit as possible in the names (e.g. `target_size_bits`).

Is this a good call? Or is it too confusing for something we don't really use?